app/src/bcsc-theme/hooks/useSystemChecks.tsx

```diff
@@ -33,48 +37,60 @@ export const useSystemChecks = (scope: SystemCheckScope) => {
   const { client, isClientReady } = useBCSCApiClientState()
   const configApi = useConfigApi(client as BCSCApiClient)
   const tokenApi = useTokenApi(client as BCSCApiClient)
-  const [loading, setLoading] = useState(false)
   const [logger] = useServices([TOKENS.UTIL_LOGGER])
-  const startupCheckRef = useRef<boolean>(false)
+  const navigation = useNavigation<InternetStatusStackNavigation>()
+  const isNavigationReady = useNavigationState((state) => Boolean(state && !state.stale))
+  const ranSystemChecksRef = useRef(false)
+
+  // Internet connectivity event listener
+  useEventListener(() => {
+    return NetInfo.addEventListener(async (netInfo) => {
+      await runSystemChecks([new InternetStatusSystemCheck(netInfo, navigation, logger)])
+    })
+  }, scope === SystemCheckScope.STARTUP && isNavigationReady)
 
   /**
    * Checks to run on app startup to ensure system is operational.
    */
   useEffect(() => {
-    const asyncEffect = async () => {
-      if (startupCheckRef.current || !isClientReady || !client) {
+    const runChecksByScope = async () => {
+      if (ranSystemChecksRef.current || !isClientReady || !client) {
         return
       }
 
-      setLoading(true)
-      startupCheckRef.current = true
-
-      let systemChecks: SystemCheckStrategy[] = []
+      ranSystemChecksRef.current = true
 
       const utils = { dispatch, translation: t, logger }
 
-      // Checks to run on app startup (root stack)
-      if (scope === SystemCheckScope.STARTUP) {
-        systemChecks = [new ServerStatusSystemCheck(configApi.getServerStatus, utils)]
-      }
-
-      // Checks to run on main stack (verified users)
-      if (scope === SystemCheckScope.MAIN_STACK) {
-        const getIdToken = () => tokenApi.getCachedIdTokenMetadata({ refreshCache: true })
-        systemChecks = [new DeviceCountSystemCheck(getIdToken, utils)]
-      }
-
       try {
-        await runSystemChecks(systemChecks)
+        // Checks to run once on app startup (root stack)
+        if (scope === SystemCheckScope.STARTUP) {
+          await runSystemChecks([new ServerStatusSystemCheck(configApi.getServerStatus, utils)])
+        }
+
+        // Checks to run once on main stack (verified users)
+        if (scope === SystemCheckScope.MAIN_STACK) {
+          const getIdToken = () => tokenApi.getCachedIdTokenMetadata({ refreshCache: false })
+          await runSystemChecks([new DeviceCountSystemCheck(getIdToken, utils)])
+        }
       } catch (error) {
         logger.error(`System checks failed: ${(error as Error).message}`)
-      } finally {
-        setLoading(false)
+        // QUESTION (MD): Should we reset this to allow re-running on next effect?
```

Comment:
I don't think so for this one, would probably be too many API calls

Reply:
I'll remove this on my next PR.